# Australian Emergency Department Data Profiling


## Purpose

This notebook examines the structure, coverage, data types, reporting levels and data-quality issues in the AIHW emergency department extract before cleaning and modelling.

In [133]:
from pathlib import Path

import pandas as pd

In [134]:
file_path = Path("../data/raw/myhosp-emergency-department-data-extract.xlsx")

if not file_path.exists():
    raise FileNotFoundError(f"Data not found: {file_path.resolve()}")

file_path

PosixPath('../data/raw/myhosp-emergency-department-data-extract.xlsx')

In [135]:
data = pd.ExcelFile(file_path)

data.sheet_names

['Patients seen on time',
 'Time in ED - within 4 hrs',
 'Time in ED',
 'Presentations']

In [136]:
# Inspect the first 15 row of each data sheet
for sheet_name in data.sheet_names:
    preview = pd.read_excel(
        file_path,
        sheet_name=sheet_name,
        header=None,
        nrows=30,
    )

    print(f"{sheet_name}: {preview.shape[0]} preview rows, {preview.shape[1]} columns")
    display(preview)

Patients seen on time: 30 preview rows, 11 columns


,0,1,2,3,4,5,6,7,8,9,10
0,NaN,www.aihw.gov.au\nAustralian Institute of Healt...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,Time spent in emergency departments,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Measure: Percentage of patients seen on time,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,"Data as of: Thursday, May 28, 2026, version: 2...",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Percentage of patients whose clinical care com...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
6,Explanatory notes,NaN,NaN,NaN,NaN,Triage 1. Resuscitation (requires immediate tr...,NaN,NaN,NaN,NaN,NaN
7,- There were no patients reported for this ind...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
8,NP Reported data did not meet the criteria to ...,NaN,NaN,NaN,NaN,Triage 2. Emergency (treatment required within...,NaN,NaN,NaN,NaN,NaN
9,NP† Results could not be accurately calculated...,NaN,NaN,NaN,NaN,Triage 3. Urgent (treatment required within 30...,NaN,NaN,NaN,NaN,NaN


Time in ED - within 4 hrs: 30 preview rows, 11 columns


,0,1,2,3,4,5,6,7,8,9,10
0,NaN,www.aihw.gov.au\nAustralian Institute of Healt...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,Time spent in emergency departments,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Measure: Percentage of patients who depart the...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,"Data as of: Thursday, May 28, 2026, version: 2...",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,The percentage of patients who departed the em...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
6,Explanatory notes,NaN,NaN,NaN,NaN,Admitted patients (Admitted from the emergency...,NaN,NaN,NaN,NaN,NaN
7,- There were no patients reported for this ind...,NaN,NaN,NaN,NaN,Discharged patients (Discharged from emergency...,NaN,NaN,NaN,NaN,NaN
8,NP Reported data did not meet the criteria to ...,NaN,NaN,NaN,NaN,Triage 1. Resuscitation (requires immediate tr...,NaN,NaN,NaN,NaN,NaN
9,NP† Results could not be accurately calculated...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


Time in ED: 30 preview rows, 13 columns


,0,1,2,3,4,5,6,7,8,9,10,11,12
0,NaN,www.aihw.gov.au\nAustralian Institute of Healt...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,Time spent in emergency departments,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Measure: Time spent in emergency department fr...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,"Data as of: Thursday, May 28, 2026, version: 2...",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,The length of time spent in the emergency depa...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
6,Explanatory notes,NaN,NaN,NaN,NaN,Admitted patients (Admitted from the emergency...,NaN,NaN,NaN,NaN,NaN,NaN,NaN
7,- There were no patients reported for this ind...,NaN,NaN,NaN,NaN,Discharged patients (Discharged from emergency...,NaN,NaN,NaN,NaN,NaN,NaN,NaN
8,NP Reported data did not meet the criteria to ...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
9,NP† Results could not be accurately calculated...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


Presentations: 30 preview rows, 6 columns


,0,1,2,3,4,5
0,NaN,www.aihw.gov.au\nAustralian Institute of Healt...,NaN,NaN,NaN,NaN
1,0,NaN,NaN,NaN,NaN,NaN
2,Measure: Total emergency department presentations,NaN,NaN,NaN,NaN,NaN
3,"Data as of: Thursday, May 28, 2026, version: 2...",NaN,NaN,NaN,NaN,NaN
4,The number of presentations (visits) to this h...,NaN,NaN,NaN,NaN,NaN
5,NaN,NaN,NaN,NaN,NaN,NaN
6,Explanatory notes,NaN,NaN,Triage 1. Resuscitation (requires immediate tr...,NaN,NaN
7,NaN,NaN,NaN,NaN,NaN,NaN
8,NaN,NaN,NaN,Triage 2. Emergency (treatment required within...,NaN,NaN
9,NaN,NaN,NaN,Triage 3. Urgent (treatment required within 30...,NaN,NaN


In [137]:
# Skip the non-structured data rows, each sheet contains only the column headers and the main data entry
sheet_config = {
    "Patients seen on time": 16,
    "Time in ED - within 4 hrs": 19,
    "Time in ED": 16,
    "Presentations": 15,
}


In [138]:
processed_data_v1 = {}

for sheet_name, rows_to_skip in sheet_config.items():
    processed_data_v1[sheet_name] = pd.read_excel(
        file_path,
        sheet_name=sheet_name,
        skiprows=rows_to_skip
    )

for sheet_name, df in processed_data_v1.items():
    print(f"\n--- {sheet_name} ---")
    print(df.shape)
    display(df.head())



--- Patients seen on time ---
(21972, 11)


,Reporting unit,Reporting unit type,State,Peer group,Year,Triage category,Number of presentations,Unnamed: 7,Percentage of patients seen on time,Unnamed: 9,Peer group average
0,National,National,NAT,NaN,2011–12,Emergency,644112,NaN,0.8,NaN,NaN
1,National,National,NAT,NaN,2012–13,Emergency,710966,NaN,0.82,NaN,NaN
2,National,National,NAT,NaN,2013–14,Emergency,781363,NaN,0.82,NaN,NaN
3,National,National,NAT,NaN,2014–15,Emergency,839117,NaN,0.79,NaN,NaN
4,National,National,NAT,NaN,2015–16,Emergency,895191,NaN,0.77,NaN,NaN



--- Time in ED - within 4 hrs ---
(36289, 11)


,Reporting unit,Reporting unit type,State,Peer group,Year,Patient cohort,Number of presentations,Unnamed: 7,Percentage who depart ED within 4 hrs,Unnamed: 9,Peer group average
0,National,National,NAT,NaN,2011–12,All patients,6547342,NaN,0.64,NaN,NaN
1,National,National,NAT,NaN,2012–13,All patients,6717067,NaN,0.67,NaN,NaN
2,National,National,NAT,NaN,2013–14,All patients,7195903,NaN,0.73,NaN,NaN
3,National,National,NAT,NaN,2014–15,All patients,7366442,NaN,0.73,NaN,NaN
4,National,National,NAT,NaN,2015–16,All patients,7465869,NaN,0.73,NaN,NaN



--- Time in ED ---
(13611, 13)


,Reporting unit,Reporting unit type,State,Peer group,Year,Patient cohort,Number of presentations,Unnamed: 7,Median time,Unnamed: 9,Time until most patients (90%) depart ED,Unnamed: 11,Peer group average (90%)
0,National,National,NAT,NaN,2011–12,All patients,6547342,NaN,2 hrs 58 mins,NaN,8 hrs 28 mins,NaN,NaN
1,National,National,NAT,NaN,2012–13,All patients,6717067,NaN,2 hrs 53 mins,NaN,7 hrs 56 mins,NaN,NaN
2,National,National,NAT,NaN,2013–14,All patients,7195903,NaN,2 hrs 40 mins,NaN,7 hrs 5 mins,NaN,NaN
3,National,National,NAT,NaN,2014–15,All patients,7366442,NaN,2 hrs 41 mins,NaN,7 hrs 2 mins,NaN,NaN
4,National,National,NAT,NaN,2015–16,All patients,7465869,NaN,2 hrs 44 mins,NaN,6 hrs 53 mins,NaN,NaN



--- Presentations ---
(22678, 6)


,Reporting unit,Reporting unit type,State,Year,Triage category,Number of presentations
0,National,National,NAT,2011–12,Emergency,647788
1,National,National,NAT,2012–13,Emergency,714124
2,National,National,NAT,2013–14,Emergency,785791
3,National,National,NAT,2014–15,Emergency,843443
4,National,National,NAT,2015–16,Emergency,899906


In [139]:
# Data Inspection

for sheet_name, df in processed_data_v1.items():
    print(f"\n{'=' * 60}")
    print(f"Sheet: {sheet_name}")
    print(f"{'=' * 60}")

    print("\nShape:")
    print(df.shape)

    print("\nColumns:")
    print(df.columns.tolist())

    print("\nData types:")
    print(df.dtypes)

    print("\nMissing values:")
    print(df.isna().sum())

    print("\nUnique values per column:")
    print(df.nunique(dropna=False))

    print("\nFirst 5 rows:")
    display(df.head())


Sheet: Patients seen on time

Shape:
(21972, 11)

Columns:
['Reporting unit', 'Reporting unit type', 'State', 'Peer group', 'Year', 'Triage category', 'Number of presentations', 'Unnamed: 7', 'Percentage of patients seen on time', 'Unnamed: 9', 'Peer group average']

Data types:
Reporting unit                             str
Reporting unit type                        str
State                                      str
Peer group                                 str
Year                                       str
Triage category                            str
Number of presentations                 object
Unnamed: 7                             float64
Percentage of patients seen on time     object
Unnamed: 9                                 str
Peer group average                      object
dtype: object

Missing values:
Reporting unit                             0
Reporting unit type                        0
State                                      0
Peer group                          

,Reporting unit,Reporting unit type,State,Peer group,Year,Triage category,Number of presentations,Unnamed: 7,Percentage of patients seen on time,Unnamed: 9,Peer group average
0,National,National,NAT,NaN,2011–12,Emergency,644112,NaN,0.8,NaN,NaN
1,National,National,NAT,NaN,2012–13,Emergency,710966,NaN,0.82,NaN,NaN
2,National,National,NAT,NaN,2013–14,Emergency,781363,NaN,0.82,NaN,NaN
3,National,National,NAT,NaN,2014–15,Emergency,839117,NaN,0.79,NaN,NaN
4,National,National,NAT,NaN,2015–16,Emergency,895191,NaN,0.77,NaN,NaN



Sheet: Time in ED - within 4 hrs

Shape:
(36289, 11)

Columns:
['Reporting unit', 'Reporting unit type', 'State', 'Peer group', 'Year', 'Patient cohort', 'Number of presentations', 'Unnamed: 7', 'Percentage who depart ED within 4 hrs', 'Unnamed: 9', 'Peer group average']

Data types:
Reporting unit                               str
Reporting unit type                          str
State                                        str
Peer group                                   str
Year                                         str
Patient cohort                               str
Number of presentations                   object
Unnamed: 7                               float64
Percentage who depart ED within 4 hrs     object
Unnamed: 9                                   str
Peer group average                        object
dtype: object

Missing values:
Reporting unit                               0
Reporting unit type                          0
State                                        0
Pee

,Reporting unit,Reporting unit type,State,Peer group,Year,Patient cohort,Number of presentations,Unnamed: 7,Percentage who depart ED within 4 hrs,Unnamed: 9,Peer group average
0,National,National,NAT,NaN,2011–12,All patients,6547342,NaN,0.64,NaN,NaN
1,National,National,NAT,NaN,2012–13,All patients,6717067,NaN,0.67,NaN,NaN
2,National,National,NAT,NaN,2013–14,All patients,7195903,NaN,0.73,NaN,NaN
3,National,National,NAT,NaN,2014–15,All patients,7366442,NaN,0.73,NaN,NaN
4,National,National,NAT,NaN,2015–16,All patients,7465869,NaN,0.73,NaN,NaN



Sheet: Time in ED

Shape:
(13611, 13)

Columns:
['Reporting unit', 'Reporting unit type', 'State', 'Peer group', 'Year', 'Patient cohort', 'Number of presentations', 'Unnamed: 7', 'Median time', 'Unnamed: 9', 'Time until most patients (90%) depart ED', 'Unnamed: 11', 'Peer group average (90%)']

Data types:
Reporting unit                                  str
Reporting unit type                             str
State                                           str
Peer group                                      str
Year                                            str
Patient cohort                                  str
Number of presentations                      object
Unnamed: 7                                  float64
Median time                                     str
Unnamed: 9                                      str
Time until most patients (90%) depart ED        str
Unnamed: 11                                     str
Peer group average (90%)                        str
dtype: object


,Reporting unit,Reporting unit type,State,Peer group,Year,Patient cohort,Number of presentations,Unnamed: 7,Median time,Unnamed: 9,Time until most patients (90%) depart ED,Unnamed: 11,Peer group average (90%)
0,National,National,NAT,NaN,2011–12,All patients,6547342,NaN,2 hrs 58 mins,NaN,8 hrs 28 mins,NaN,NaN
1,National,National,NAT,NaN,2012–13,All patients,6717067,NaN,2 hrs 53 mins,NaN,7 hrs 56 mins,NaN,NaN
2,National,National,NAT,NaN,2013–14,All patients,7195903,NaN,2 hrs 40 mins,NaN,7 hrs 5 mins,NaN,NaN
3,National,National,NAT,NaN,2014–15,All patients,7366442,NaN,2 hrs 41 mins,NaN,7 hrs 2 mins,NaN,NaN
4,National,National,NAT,NaN,2015–16,All patients,7465869,NaN,2 hrs 44 mins,NaN,6 hrs 53 mins,NaN,NaN



Sheet: Presentations

Shape:
(22678, 6)

Columns:
['Reporting unit', 'Reporting unit type', 'State', 'Year', 'Triage category', 'Number of presentations']

Data types:
Reporting unit                str
Reporting unit type           str
State                         str
Year                          str
Triage category               str
Number of presentations    object
dtype: object

Missing values:
Reporting unit             0
Reporting unit type        0
State                      0
Year                       0
Triage category            0
Number of presentations    0
dtype: int64

Unique values per column:
Reporting unit               397
Reporting unit type            4
State                          9
Year                          14
Triage category                5
Number of presentations    10593
dtype: int64

First 5 rows:


,Reporting unit,Reporting unit type,State,Year,Triage category,Number of presentations
0,National,National,NAT,2011–12,Emergency,647788
1,National,National,NAT,2012–13,Emergency,714124
2,National,National,NAT,2013–14,Emergency,785791
3,National,National,NAT,2014–15,Emergency,843443
4,National,National,NAT,2015–16,Emergency,899906


In [140]:
# Assign names for each data sheet

presentations = processed_data_v1["Presentations"]
seen_on_time = processed_data_v1["Patients seen on time"]
within_4hrs = processed_data_v1["Time in ED - within 4 hrs"]
time_in_ed = processed_data_v1["Time in ED"]



In [141]:
## Check data quality 

# 1. presentations 

# categorical inspection
for col in [
    "Reporting unit type",
    "State",
    "Year",
    "Triage category"
]:
    print(f"\n--- {col} ---")
    print(presentations[col].value_counts(dropna=False))


--- Reporting unit type ---
Reporting unit type
Hospital                  19375
Local Hospital Network     2678
State                       555
National                     70
Name: count, dtype: int64

--- State ---
State
NSW    12170
Vic     3803
Qld     2370
WA      1720
SA      1460
NT       510
Tas      385
ACT      190
NAT       70
Name: count, dtype: int64

--- Year ---
Year
2021–22    1904
2020–21    1894
2022–23    1894
2023–24    1894
2024–25    1894
2018–19    1879
2019–20    1874
2014–15    1480
2016–17    1475
2017–18    1470
2013–14    1465
2015–16    1460
2012–13    1050
2011–12    1045
Name: count, dtype: int64

--- Triage category ---
Triage category
Emergency        4537
Non-Urgent       4537
Semi-Urgent      4537
Urgent           4537
Resuscitation    4530
Name: count, dtype: int64


In [142]:
# inspect the measure column
presentations["Number of presentations"].value_counts(
    dropna=False
).head(30)

Number of presentations
<5    616
0     256
5      92
8      82
7      69
6      68
9      60
12     59
10     49
30     40
13     36
14     36
42     34
71     33
29     32
16     32
11     32
23     31
15     31
18     31
38     31
25     30
22     30
48     29
37     29
32     29
20     29
17     29
34     28
40     27
Name: count, dtype: int64

In [143]:
non_numeric_values = presentations.loc[
    presentations["Number of presentations"].notna()
    & pd.to_numeric(
        presentations["Number of presentations"],
        errors="coerce"
    ).isna(),
    "Number of presentations"
]

non_numeric_values.value_counts(dropna=False)

Number of presentations
<5    616
Name: count, dtype: int64

In [144]:
# Missing values in the measure column
missing_presentations = presentations[
    presentations["Number of presentations"].isna()
]

print("Missing rows:", len(missing_presentations))

Missing rows: 0


Number of presentations contains no missing values. However, 616 records are reported as <5, which prevents the column from being stored as numeric. These small-count values will require explicit treatment during cleaning.

In [145]:
# Check if there is any duplications
candidate_key = [
    "Reporting unit",
    "Reporting unit type",
    "State",
    "Year",
    "Triage category"
]

presentations.duplicated(
    subset=candidate_key,
    keep=False
).sum()



0

In [146]:
presentations.duplicated().sum()

0

In [147]:
# Insepct the <5 records

small_counts = presentations[
    presentations["Number of presentations"] == "<5"
]

small_counts["Reporting unit type"].value_counts()

Reporting unit type
Hospital    616
Name: count, dtype: int64

In [148]:
# Check the category that these small counts belong to
small_counts["Triage category"].value_counts()

Triage category
Resuscitation    563
Emergency         21
Urgent            19
Semi-Urgent        7
Non-Urgent         6
Name: count, dtype: int64

In [149]:
small_counts["Triage category"].value_counts()

Triage category
Resuscitation    563
Emergency         21
Urgent            19
Semi-Urgent        7
Non-Urgent         6
Name: count, dtype: int64

In [150]:
# Also check zero values
zero_counts = presentations[
    presentations["Number of presentations"] == 0
]

len(zero_counts)

256

The Presentations table has no duplicate rows and the candidate key (Reporting unit, Reporting unit type, State, Year, Triage category) uniquely identifies each record. The Number of presentations field contains 616 suppressed small counts (<5), all at hospital level and predominantly in the Resuscitation category. There are also 256 explicit zero values, which should remain distinct from suppressed small counts.

In [151]:
zero_counts["Reporting unit type"].value_counts()

Reporting unit type
Hospital    256
Name: count, dtype: int64

In [152]:
zero_counts["Triage category"].value_counts()

Triage category
Resuscitation    233
Emergency         15
Urgent             6
Semi-Urgent        2
Name: count, dtype: int64

In [153]:
presentations["Year"].value_counts().sort_index()

Year
2011–12    1045
2012–13    1050
2013–14    1465
2014–15    1480
2015–16    1460
2016–17    1475
2017–18    1470
2018–19    1879
2019–20    1874
2020–21    1894
2021–22    1904
2022–23    1894
2023–24    1894
2024–25    1894
Name: count, dtype: int64

The number of records varies across financial years, indicating changes in reporting coverage or participating reporting units over time. Longitudinal comparisons at hospital level should therefore account for differences in reporting coverage.

In [154]:
# For seen_on_time data, first inspect the two suspicious empty column
seen_on_time[["Unnamed: 7", "Unnamed: 9"]].notna().sum()

Unnamed: 7      0
Unnamed: 9    385
dtype: int64

In [155]:
seen_on_time[["Unnamed: 7", "Unnamed: 9"]].isna().all()

Unnamed: 7     True
Unnamed: 9    False
dtype: bool

In [156]:
seen_on_time["Unnamed: 9"].value_counts(dropna=False)

Unnamed: 9
NaN    21587
*        385
Name: count, dtype: int64

In [157]:
seen_on_time.loc[
    seen_on_time["Unnamed: 9"].notna(),
    [
        "Reporting unit",
        "Reporting unit type",
        "State",
        "Year",
        "Triage category",
        "Percentage of patients seen on time",
        "Unnamed: 9",
    ]
].head(30)

,Reporting unit,Reporting unit type,State,Year,Triage category,Percentage of patients seen on time,Unnamed: 9
155,Armidale Hospital,Hospital,NSW,2012–13,Non-Urgent,0.93,*
158,Armidale Hospital,Hospital,NSW,2015–16,Non-Urgent,0.94,*
159,Armidale Hospital,Hospital,NSW,2016–17,Non-Urgent,0.94,*
160,Armidale Hospital,Hospital,NSW,2017–18,Non-Urgent,0.93,*
174,Armidale Hospital,Hospital,NSW,2017–18,Resuscitation,1,*
227,Auburn Hospital,Hospital,NSW,2014–15,Non-Urgent,0.95,*
228,Auburn Hospital,Hospital,NSW,2015–16,Non-Urgent,0.91,*
229,Auburn Hospital,Hospital,NSW,2016–17,Non-Urgent,0.91,*
230,Auburn Hospital,Hospital,NSW,2017–18,Non-Urgent,0.91,*
231,Auburn Hospital,Hospital,NSW,2018–19,Non-Urgent,0.87,*


In [158]:
# inspect the categorical columns of seen_on_time 

for col in [
    "Reporting unit type",
    "State",
    "Peer group",
    "Year",
    "Triage category"
]:
    print(f"\n--- {col} ---")
    print(seen_on_time[col].value_counts(dropna=False))


--- Reporting unit type ---
Reporting unit type
Hospital                  18717
Local Hospital Network     2630
State                       555
National                     70
Name: count, dtype: int64

--- State ---
State
NSW    11499
Vic     3775
Qld     2370
WA      1713
SA      1460
NT       510
Tas      385
ACT      190
NAT       70
Name: count, dtype: int64

--- Peer group ---
Peer group
Unpeered                         4852
Small hospitals                  3915
NaN                              2700
Large metropolitan hospitals     2129
Major hospitals                  2104
Large regional hospitals         2040
Medium metropolitan hospitals    1752
Medium regional hospitals        1475
National                          555
Children's hospitals              450
Name: count, dtype: int64

--- Year ---
Year
2018–19    1879
2023–24    1787
2024–25    1787
2021–22    1779
2022–23    1776
2020–21    1768
2019–20    1751
2014–15    1480
2016–17    1475
2017–18    1470
2013–14    1465
2

In [159]:
# Inspect the performance measure
for col in [
    "Number of presentations",
    "Percentage of patients seen on time", 
    "Peer group average"
]:
    print(f"\n--- {col} ---")
    print(seen_on_time[col].value_counts(dropna=False).head(30)) 


--- Number of presentations ---
Number of presentations
<5    329
0     146
5      54
12     53
10     51
8      41
6      41
15     40
7      38
13     38
22     36
28     35
70     35
14     35
17     34
11     34
37     33
40     31
66     31
26     31
32     31
16     31
31     30
30     29
33     29
18     29
47     28
35     28
42     28
61     28
Name: count, dtype: int64

--- Percentage of patients seen on time ---
Percentage of patients seen on time
1       4201
0.99     788
0.98     665
0.97     588
0.96     572
NP†      562
0.94     550
0.92     530
0.93     518
0.95     507
NP       495
0.91     485
0.9      477
0.85     445
0.89     440
0.87     428
0.82     418
0.86     417
0.83     415
0.84     413
0.88     411
0.8      404
0.79     389
0.78     389
0.81     377
0.77     371
0.75     347
0.76     341
0.74     322
0.73     305
Name: count, dtype: int64

--- Peer group average ---
Peer group average
Not peered    4852
NaN           2700
1             2375
0.92           6

In [160]:
seen_on_time[ 
    pd.to_numeric( 
        seen_on_time["Number of presentations"],
        errors="coerce" 
    ).isna() 
][ 
    "Number of presentations"
].value_counts(dropna=False)  

Number of presentations
<5    329
Name: count, dtype: int64

In [161]:
seen_on_time[
    pd.to_numeric(
        seen_on_time["Percentage of patients seen on time"],
        errors="coerce"
    ).isna()
][
    "Percentage of patients seen on time"
].value_counts(dropna=False)

Percentage of patients seen on time
NP†    562
NP     495
-      146
Name: count, dtype: int64

In [162]:
# Inspect the rows containing NP, NP+ and - 
special_performance = seen_on_time[
    seen_on_time["Percentage of patients seen on time"].isin(
        ["NP", "NP†", "–"]
    )
]

special_performance[
    [
        "Reporting unit type",
        "State",
        "Year",
        "Triage category",
        "Percentage of patients seen on time",
        "Unnamed: 9",
    ]
].head(30)

,Reporting unit type,State,Year,Triage category,Percentage of patients seen on time,Unnamed: 9
224,Hospital,NSW,2011–12,Non-Urgent,NP†,NaN
225,Hospital,NSW,2012–13,Non-Urgent,NP†,NaN
226,Hospital,NSW,2013–14,Non-Urgent,NP†,NaN
293,Hospital,NSW,2011–12,Non-Urgent,NP†,NaN
294,Hospital,NSW,2012–13,Non-Urgent,NP†,NaN
295,Hospital,NSW,2013–14,Non-Urgent,NP†,NaN
296,Hospital,NSW,2014–15,Non-Urgent,NP†,NaN
297,Hospital,NSW,2015–16,Non-Urgent,NP†,NaN
298,Hospital,NSW,2016–17,Non-Urgent,NP†,NaN
299,Hospital,NSW,2017–18,Non-Urgent,NP†,NaN


In [163]:
pd.crosstab(
    special_performance["Reporting unit type"],
    special_performance["Percentage of patients seen on time"]
)

Percentage of patients seen on time,NP,NP†
Reporting unit type,,
Hospital,485,560
Local Hospital Network,10,0
State,0,2


In [164]:
pd.crosstab(
    special_performance["Triage category"],
    special_performance["Percentage of patients seen on time"]
)

Percentage of patients seen on time,NP,NP†
Triage category,,
Emergency,25,86
Non-Urgent,2,229
Resuscitation,448,39
Semi-Urgent,5,113
Urgent,15,95


In [165]:
candidate_key_seen_on_time = [
    "Reporting unit",
    "Reporting unit type",
    "State",
    "Peer group",
    "Year",
    "Triage category",
]

seen_on_time.duplicated(
    subset=candidate_key_seen_on_time,
    keep=False
).sum()

0

In [166]:
seen_on_time.duplicated().sum()

0

In [167]:
seen_on_time.loc[
    seen_on_time["Unnamed: 9"] == "*",
    "Percentage of patients seen on time"
].apply(lambda x: pd.to_numeric(x, errors="coerce")).isna().sum()

0

In [168]:
seen_on_time[
    pd.to_numeric(
        seen_on_time["Peer group average"],
        errors="coerce"
    ).isna()
][
    "Peer group average"
].value_counts(dropna=False)

Peer group average
Not peered    4852
NaN           2700
Name: count, dtype: int64

In [169]:
pd.crosstab(
    seen_on_time["Reporting unit type"],
    seen_on_time["Peer group average"].isna()
)

Peer group average,False,True
Reporting unit type,,
Hospital,18717,0
Local Hospital Network,0,2630
National,0,70
State,555,0


In [170]:
seen_on_time.loc[
    seen_on_time["Peer group average"] == "Not peered",
    "Reporting unit type"
].value_counts()

Reporting unit type
Hospital    4852
Name: count, dtype: int64

In [171]:
seen_on_time.loc[
    seen_on_time["Peer group average"].isna(),
    "Reporting unit type"
].value_counts()

Reporting unit type
Local Hospital Network    2630
National                    70
Name: count, dtype: int64

In [172]:
pd.crosstab(
    seen_on_time["Peer group"],
    seen_on_time["Peer group average"] == "Not peered"
)

Peer group average,False,True
Peer group,,
Children's hospitals,450,0
Large metropolitan hospitals,2129,0
Large regional hospitals,2040,0
Major hospitals,2104,0
Medium metropolitan hospitals,1752,0
Medium regional hospitals,1475,0
National,555,0
Small hospitals,3915,0
Unpeered,0,4852


The patients seen on time table contains no duplicate rows, and the candidate key (Reporting unit, Reporting unit type, State, Peer group, Year, Triage category) uniquely identifies each record. The Percentage of patients seen on time field includes publication-status codes (NP, NP†, –) that require preservation during cleaning. A separate * flag marks 385 numeric results that should be interpreted with caution due to 5–10% missing or invalid time data. Peer-group values also contain structural states such as Unpeered and Not peered, which should not be treated as ordinary missing values.

In [173]:
# For within_4_hrs

for col in [
    "Reporting unit type",
    "State",
    "Peer group",
    "Year",
    "Patient cohort"
]:
    print(f"\n--- {col} ---")
    print(within_4hrs[col].value_counts(dropna=False))


--- Reporting unit type ---
Reporting unit type
Hospital                  31000
Local Hospital Network     4289
State                       888
National                    112
Name: count, dtype: int64

--- State ---
State
NSW    19472
Vic     6089
Qld     3792
WA      2752
SA      2336
NT       816
Tas      616
ACT      304
NAT      112
Name: count, dtype: int64

--- Peer group ---
Peer group
Unpeered                         8688
Small hospitals                  6352
NaN                              4401
Large metropolitan hospitals     3408
Major hospitals                  3368
Large regional hospitals         3264
Medium metropolitan hospitals    2832
Medium regional hospitals        2368
National                          888
Children's hospitals              720
Name: count, dtype: int64

--- Year ---
Year
2021–22    3047
2020–21    3031
2022–23    3031
2023–24    3031
2024–25    3031
2018–19    3007
2019–20    2999
2014–15    2368
2016–17    2360
2017–18    2352
2013–14    2344
2

In [174]:
for col in [
    "Number of presentations",
    "Percentage who depart ED within 4 hrs",
    "Peer group average"
]:
    print(f"\n--- {col} ---")
    print(within_4hrs[col].value_counts(dropna=False).head(30))


--- Number of presentations ---
Number of presentations
<5    653
0     373
5      96
8      89
7      73
6      69
9      66
12     65
10     52
30     50
71     39
15     38
13     38
18     37
14     37
48     36
42     36
29     36
34     36
23     36
16     36
20     35
11     35
25     34
38     34
27     34
17     33
33     32
22     32
37     31
Name: count, dtype: int64

--- Percentage who depart ED within 4 hrs ---
Percentage who depart ED within 4 hrs
NP      1067
0.93     854
0.95     844
0.96     824
0.94     819
0.92     818
0.97     795
0.98     793
0.91     750
0.99     742
0.9      724
0.88     705
0.89     692
0.87     659
0.85     648
0.86     642
0.84     636
0.82     596
0.83     586
0.81     577
0.8      558
0.79     523
0.77     519
0.76     519
0.75     510
1        497
0.78     493
0.74     486
0.72     479
0.71     477
Name: count, dtype: int64

--- Peer group average ---
Peer group average
Not peered    8688
NaN           4401
0.65           713
0.78        

In [175]:
within_4hrs[
    pd.to_numeric(
        within_4hrs["Number of presentations"],
        errors="coerce"
    ).isna()
][
    "Number of presentations"
].value_counts(dropna=False)

Number of presentations
<5    653
Name: count, dtype: int64

In [176]:
within_4hrs[
    pd.to_numeric(
        within_4hrs["Percentage who depart ED within 4 hrs"],
        errors="coerce"
    ).isna()
][
    "Percentage who depart ED within 4 hrs"
].value_counts(dropna=False)

Percentage who depart ED within 4 hrs
NP     1067
-       373
NP†     189
Name: count, dtype: int64

In [203]:
special_values = within_4hrs[
    pd.to_numeric(
        within_4hrs["Percentage who depart ED within 4 hrs"],
        errors="coerce"
    ).isna()
]["Percentage who depart ED within 4 hrs"].dropna().unique()

for value in special_values:
    print(repr(value))

'NP'
'-'
'NP†'


In [177]:
within_4hrs[
    pd.to_numeric(
        within_4hrs["Peer group average"],
        errors="coerce"
    ).isna()
][
    "Peer group average"
].value_counts(dropna=False)

Peer group average
Not peered    8688
NaN           4401
Name: count, dtype: int64

In [178]:
# Check unnamed columns
within_4hrs[["Unnamed: 7", "Unnamed: 9"]].notna().sum()

Unnamed: 7      0
Unnamed: 9    306
dtype: int64

In [179]:
within_4hrs["Unnamed: 9"].value_counts(dropna=False)

Unnamed: 9
NaN    35983
*        306
Name: count, dtype: int64

In [180]:
within_4hrs.loc[
    within_4hrs["Unnamed: 9"].notna(),
    [
        "Reporting unit",
        "Reporting unit type",
        "State",
        "Year",
        "Patient cohort",
        "Percentage who depart ED within 4 hrs",
        "Unnamed: 9",
    ]
].head(30)

,Reporting unit,Reporting unit type,State,Year,Patient cohort,Percentage who depart ED within 4 hrs,Unnamed: 9
622,Balranald Multi Purpose Service,Hospital,NSW,2015–16,Semi-Urgent,0.96,*
634,Balranald Multi Purpose Service,Hospital,NSW,2015–16,Subsequently admitted patients,0.98,*
645,Balranald Multi Purpose Service,Hospital,NSW,2014–15,Urgent,0.93,*
646,Balranald Multi Purpose Service,Hospital,NSW,2015–16,Urgent,0.95,*
961,Barraba Multi Purpose Service,Hospital,NSW,2012–13,All patients,0.98,*
962,Barraba Multi Purpose Service,Hospital,NSW,2013–14,All patients,0.99,*
963,Barraba Multi Purpose Service,Hospital,NSW,2014–15,All patients,0.98,*
975,Barraba Multi Purpose Service,Hospital,NSW,2012–13,Emergency,0.91,*
990,Barraba Multi Purpose Service,Hospital,NSW,2013–14,Non-Urgent,1,*
991,Barraba Multi Purpose Service,Hospital,NSW,2014–15,Non-Urgent,1,*


In [181]:
within_4hrs[
    pd.to_numeric(
        within_4hrs["Peer group average"],
        errors="coerce"
    ).isna()
]["Peer group average"].value_counts(dropna=False)

Peer group average
Not peered    8688
NaN           4401
Name: count, dtype: int64

In [182]:
pd.crosstab(
    within_4hrs["Reporting unit type"],
    within_4hrs["Peer group average"].isna()
)

Peer group average,False,True
Reporting unit type,,
Hospital,31000,0
Local Hospital Network,0,4289
National,0,112
State,888,0


In [183]:
within_4hrs.loc[
    within_4hrs["Peer group average"] == "Not peered",
    "Reporting unit type"
].value_counts() 

Reporting unit type
Hospital    8688
Name: count, dtype: int64

In [184]:
pd.crosstab(
    within_4hrs["Peer group"],
    within_4hrs["Peer group average"] == "Not peered"
) 

Peer group average,False,True
Peer group,,
Children's hospitals,720,0
Large metropolitan hospitals,3408,0
Large regional hospitals,3264,0
Major hospitals,3368,0
Medium metropolitan hospitals,2832,0
Medium regional hospitals,2368,0
National,888,0
Small hospitals,6352,0
Unpeered,0,8688


In [185]:
candidate_key_within_4hrs = [
    "Reporting unit",
    "Reporting unit type",
    "State",
    "Peer group",
    "Year",
    "Patient cohort",
]

within_4hrs.duplicated(
    subset=candidate_key_within_4hrs,
    keep=False
).sum()

0

In [186]:
within_4hrs.duplicated().sum()

0

Key findings:

- Unnamed: 7 is fully empty -> safe to drop later.
- Unnamed: 9 contains only * flags -> same caution meaning as the previous - sheet.
- -Number of presentations has 653 <5 values -> preserve as small-count - suppression.
- Percentage who depart ED within 4 hrs contains NP, NP†, and – -> same publication-status logic as before.
- Peer group average is structurally missing for Local Hospital Network and National rows.
- Not peered occurs only for hospital rows.
- Peer group = Unpeered maps exactly to Peer group average = Not peered.
- Candidate key duplicates = 0.
- Exact duplicates = 0.


In [187]:
# For time_in_ed

for col in [
    "Median time",
    "Time until most patients (90%) depart ED",
    "Peer group average (90%)" 
]: 
    print(f"\n--- {col} ---")
    print(time_in_ed[col].value_counts(dropna=False).head(30))


--- Median time ---
Median time
0 hrs 30 mins    118
-                117
0 hrs 45 mins    106
1 hrs 0 mins      99
0 hrs 50 mins     92
1 hrs 15 mins     89
0 hrs 40 mins     87
1 hrs 30 mins     86
0 hrs 55 mins     85
1 hrs 10 mins     82
2 hrs 0 mins      79
1 hrs 40 mins     75
1 hrs 5 mins      73
1 hrs 20 mins     73
0 hrs 25 mins     73
2 hrs 59 mins     70
2 hrs 30 mins     70
2 hrs 38 mins     70
2 hrs 10 mins     69
2 hrs 25 mins     69
3 hrs 2 mins      69
NP                68
2 hrs 40 mins     67
3 hrs 14 mins     67
1 hrs 50 mins     66
2 hrs 44 mins     64
2 hrs 19 mins     64
2 hrs 31 mins     64
2 hrs 37 mins     64
2 hrs 54 mins     64
Name: count, dtype: int64

--- Time until most patients (90%) depart ED ---
Time until most patients (90%) depart ED
-                117
NP                68
5 hrs 0 mins      64
3 hrs 0 mins      54
4 hrs 0 mins      54
3 hrs 30 mins     53
3 hrs 50 mins     53
NP†               49
5 hrs 30 mins     48
2 hrs 30 mins     48
4 hrs 30 m

In [188]:
time_in_ed["Median time"].dropna().astype(str).head(20) 

0     2 hrs 58 mins
1     2 hrs 53 mins
2     2 hrs 40 mins
3     2 hrs 41 mins
4     2 hrs 44 mins
5     2 hrs 48 mins
6     2 hrs 53 mins
7     2 hrs 58 mins
8     2 hrs 56 mins
9      3 hrs 1 mins
10    3 hrs 20 mins
11    3 hrs 39 mins
12    3 hrs 42 mins
13    3 hrs 48 mins
14    2 hrs 18 mins
15    2 hrs 19 mins
16     2 hrs 7 mins
17     2 hrs 8 mins
18    2 hrs 13 mins
19    2 hrs 16 mins
Name: Median time, dtype: str

In [202]:
time_in_ed["Patient cohort"].value_counts(dropna=False)

Patient cohort
All patients                          4537
Not subsequently admitted patients    4537
Subsequently admitted patients        4537
Name: count, dtype: int64

In [193]:
time_in_ed[
    ["Unnamed: 7", "Unnamed: 9", "Unnamed: 11"]
].notna().sum()

Unnamed: 7      0
Unnamed: 9     94
Unnamed: 11    94
dtype: int64

In [194]:
for col in ["Unnamed: 7", "Unnamed: 9", "Unnamed: 11"]:
    print(f"\n--- {col} ---")
    print(time_in_ed[col].value_counts(dropna=False))


--- Unnamed: 7 ---
Unnamed: 7
NaN    13611
Name: count, dtype: int64

--- Unnamed: 9 ---
Unnamed: 9
NaN    13517
*         94
Name: count, dtype: int64

--- Unnamed: 11 ---
Unnamed: 11
NaN    13517
*         94
Name: count, dtype: int64


In [200]:
(
    time_in_ed["Unnamed: 9"].fillna("")
    == time_in_ed["Unnamed: 11"].fillna("")
).all()

True

In [201]:
non_numeric_presentations_time = time_in_ed.loc[
    time_in_ed["Number of presentations"].notna()
    & pd.to_numeric(
        time_in_ed["Number of presentations"],
        errors="coerce"
    ).isna(),
    "Number of presentations"
]

non_numeric_presentations_time.value_counts(dropna=False)

Number of presentations
<5    37
Name: count, dtype: int64

In [195]:
duration_pattern = r"^\d+\s+hrs?\s+\d+\s+mins?$"

for col in [
    "Median time",
    "Time until most patients (90%) depart ED",
    "Peer group average (90%)"
]:
    special_values = time_in_ed.loc[
        time_in_ed[col].notna()
        & ~time_in_ed[col].astype(str).str.match(duration_pattern),
        col
    ]

    print(f"\n--- Special values in {col} ---")
    print(special_values.value_counts(dropna=False))


--- Special values in Median time ---
Median time
-      117
NP      68
NP†     49
Name: count, dtype: int64

--- Special values in Time until most patients (90%) depart ED ---
Time until most patients (90%) depart ED
-      117
NP      68
NP†     49
Name: count, dtype: int64

--- Special values in Peer group average (90%) ---
Peer group average (90%)
Not peered    1716
Name: count, dtype: int64


In [196]:
pd.crosstab(
    time_in_ed["Reporting unit type"],
    time_in_ed["Peer group average (90%)"].isna()
)

Peer group average (90%),False,True
Reporting unit type,,
Hospital,11625,0
Local Hospital Network,0,1611
National,0,42
State,333,0


In [197]:
time_in_ed.loc[
    time_in_ed["Peer group average (90%)"] == "Not peered",
    "Reporting unit type"
].value_counts()

Reporting unit type
Hospital    1716
Name: count, dtype: int64

In [198]:
candidate_key_time_in_ed = [
    "Reporting unit",
    "Reporting unit type",
    "State",
    "Peer group",
    "Year",
    "Patient cohort"
]

time_in_ed.duplicated(
    subset=candidate_key_time_in_ed,
    keep=False
).sum()

0

In [199]:
time_in_ed.duplicated().sum()

0

Main Findings:
- Unnamed: 7 is completely empty -> safe to drop later.
- Unnamed: 9 contains 94 * flags.
- Unnamed: 11 also contains 94 * flags.
- Median time and Time until most patients (90%) depart ED use consistent duration strings such as 2 hrs 58 mins.
- Both duration fields also contain the publication codes –, NP, and NP†.
- Peer group average (90%) contains Not peered for 1,716 rows.
- Peer-group averages are structurally missing for National and Local Hospital Network rows.
- Not peered occurs only for hospital records.
- Candidate-key duplicates = 0.
- Exact duplicates = 0.